src/format/scan.rs

```diff
@@ -262,13 +275,30 @@ where
             Ok((b[0], b[1]))
         }
     }
-    let negative = match s.as_bytes().first() {
-        Some(&b'+') => false,
-        Some(&b'-') => true,
+    let negative = match s.chars().next() {
```

Comment:
I'd like a separate commit first that switches from iterating over bytes to characters.